pandas/core/internals/managers.py

```diff
@@ -2358,7 +2359,7 @@ def _preprocess_slice_or_indexer(
 def make_na_array(dtype: DtypeObj, shape: Shape, fill_value) -> ArrayLike:
     if isinstance(dtype, DatetimeTZDtype):
         # NB: exclude e.g. pyarrow[dt64tz] dtypes
-        i8values = np.full(shape, fill_value._value)
+        i8values = np.full(shape, NaT.value)
```

Comment:
the reason a test is failing is because in cases where fill_value is not None you need to keep it

Reply:
> the reason a test is failing is because in cases where fill_value is not None you need to keep it


```suggestion
        if fill_value is None:
            i8values = np.full(shape, NaT.value)
        else:
            i8values = np.full(shape, fill_value._value)
```
1. Do you think this change could fulfill that?
2. Depreciation warning
    I got a deprecation warning while debugging.
```sh
FutureWarning: The behavior of DataFrame concatenation with all-NA entries is deprecated. In a future version, this will no longer exclude all-NA columns when determining the result dtypes. To retain the old behavior, cast the all-NA columns to the desired dtype before the concat operation.
```
So I tried re-running the example by changing `None` dtype to `datetime64[ns, UTC]`. It worked and didn't raise any errors. 
```py
>>> data = [{'A': None}]
>>> df = pd.DataFrame(data, dtype="datetime64[ns, UTC]")
>>> data_2 = [{'A': pd.to_datetime("1990-12-20 00:00:00+00:00")}]
>>> df_2 = pd.DataFrame(data_2)
>>> # df.dtypes
>>> pd.concat([df, df_2])

# o/p
	A
0	NaT
0	1990-12-20 00:00:00+00:00
```

Reply:
Try `Timestamp(fill_value)._value`